### PR TITLE
763 admins upload corrections

### LINKED ISSUE #763: Sending corrections to students through ETD
From Grad School:

> As of now, OTD emails students their corrections in PDF, which is very manual and takes a lot of time. We would like ETD to allow us to send students our corrections through ETD so that everything is in one place.

We will need to add a field for admins to upload this document in the admin submission screen.  They can add any additional comments in the existing "Notes to Student" field.  Then, we we'll need a place for the student to download the file on the author end.

### PR DESCRIPTION
**Pull request by jlandiseigsti**
Closes #763

### CODE REVIEW

**Comment by ajkiessl** [comment]
This looks good to me so far

**Comment by jlandiseigsti** [comment]
Ok, at long last, it's out of draft mode and ready for actual reviews :D

File: db/schema.rb [hunk 1]
```diff
@@ -293,6 +301,8 @@
     t.string "lionpath_semester"
     t.string "academic_program"
     t.string "degree_checkout_status"
+    t.datetime "author_release_warning_sent_at"
+    t.datetime "acknowledgment_page_submitted_at"
```
Comment (ajkiessl) [inline db/schema.rb:305]: These will need to be removed.

**Review by ajkiessl — CHANGES_REQUESTED** [review]
We'll need some more tests here.  Here's what I'm thinking:

- Model tests for `AdminFeedbackFile`
- Tests in `AuthorAbility`to test access to `AdminFeedbackFile`
- Unit tests for the new methods in `submission.rb`
- An integration test to test uploading, linking to, and deleting the admin feedback files in the final submission form
- An integration test to test author viewing of the feedback files at the final submission and format review stages.  Also, a test somewhere to test what happens when there isn't a file

File: spec/models/submission_spec.rb [hunk 2]
```diff
@@ -886,4 +887,28 @@
...
+
+  describe "#final_submission_feedback_files?" do
+    it 'returns true if at least one admin feedback file with type final-submission' do
+      submission.admin_feedback_files.build(feedback_type: 'final-submission')
+      expect(submission).to be_final_submission_feedback_files
+    end
+
+    it 'returns false if no admin feedback file has the type of final-submission' do
+      submission.admin_feedback_files.build(feedback_type: 'not final-submission')
+      expect(submission).not_to be_final_submission_notes
```
Comment (ajkiessl) [inline spec/models/submission_spec.rb:899]: Should this be `be_final_submission_feedback_files`?

File: spec/models/submission_spec.rb [hunk 3]
```diff
@@ -886,4 +887,28 @@
...
+
+  describe "#format_review_feedback_files?" do
+    it 'returns true if at least one admin feedback file with type format-review' do
+      submission.admin_feedback_files.build(feedback_type: 'format-review')
+      expect(submission).to be_final_submission_feedback_files
+    end
+
+    it 'returns false if no admin feedback file has the type of format-review' do
+      submission.admin_feedback_files.build(feedback_type: 'not format-review')
+      expect(submission).not_to be_final_submission_notes
```
Comment (ajkiessl) [inline spec/models/submission_spec.rb:911]: This too: `be_final_submission_feedback_files`?

**Review by ajkiessl — CHANGES_REQUESTED** [review]
This looks good.  Just two minor things I found in the tests.

File: spec/models/submission_spec.rb (hunk 3 above)
Comment (jlandiseigsti) [reply spec/models/submission_spec.rb:911]: Good catch!

**Review by ajkiessl — APPROVED** [review]
lgtm

**Pull request merged**